django/views/debug.py

```diff
@@ -266,61 +265,33 @@ def __init__(self, request, exc_type, exc_value, tb, is_email=False):
             self.exc_type = type(self.exc_value)
 
-    def format_path_status(self, path):
-        if not os.path.exists(path):
-            return "File does not exist"
-        return "File exists"
-
     def get_traceback_data(self):
         """Return a dictionary containing traceback information."""
-        try:
-            default_template_engine = Engine.get_default()
-        except Exception:
-            # Since the debug view must never crash, catch all exceptions.
-            # If Django can't find a default template engine, get_default()
-            # raises ImproperlyConfigured. If some template engines fail to
-            # load, any exception may be raised.
-            default_template_engine = None
-
-        # TODO: add support for multiple template engines (#24120).
-        # TemplateDoesNotExist should carry all the information.
-        # Replaying the search process isn't a good design.
         if self.exc_type and issubclass(self.exc_type, TemplateDoesNotExist):
-            if default_template_engine is None:
-                template_loaders = []
-            else:
-                self.template_does_not_exist = True
-                self.loader_debug_info = []
-                # If Django fails in get_template_loaders, provide an empty list
-                # for the following loop to not fail.
-                try:
-                    template_loaders = default_template_engine.template_loaders
-                except Exception:
-                    template_loaders = []
-
-            for loader in template_loaders:
-                try:
-                    source_list_func = loader.get_template_sources
-                    # NOTE: This assumes exc_value is the name of the template that
-                    # the loader attempted to load.
-                    template_list = [{
-                        'name': t,
-                        'status': self.format_path_status(t),
-                    } for t in source_list_func(str(self.exc_value))]
-                except AttributeError:
-                    template_list = []
-                loader_name = loader.__module__ + '.' + loader.__class__.__name__
-                self.loader_debug_info.append({
-                    'loader': loader_name,
-                    'templates': template_list,
-                })
+            self.template_does_not_exist = True
+            postmortem = []
+
+            for engine in engines.all():
+                if hasattr(engine, 'engine'):
+                    e = engine.engine
+                else:
+                    e = engine
+
+                postmortem.append(dict(
+                    engine=engine,
+                    tried=[
+                        entry for entry in self.exc_value.tried if
+                        entry[0].loader.engine == e
+                    ],
+                ))
+            self.postmortem = postmortem
```

Comment:
This doesn't matter much. Just ensure that such edge cases don't raise exceptions.